### ISSUE TITLE
Allow for a neutrino hierarchy 

### ISSUE THREAD

**Issue by tmcclintock** [philbull/cosmoparams#1]
My understanding is that current and future surveys are/will be able to probe the neutrino mass hierarchy. It would make sense to allow for not just three massive neutrinos with identical masses.

**Comment by philbull** [comment]
I think this is a good point and bears discussion. My feeling is that we should support the "most basic" neutrino treatment that we can in the base class, to keep things simple. We want to avoid a situation where (e.g.) supernova people, who don't need to care about massive neutrinos, have to spend a lot of time writing and testing neutrino-related code to comply with the standard when they don't need to use it.

It's possible that we can support a richer description of neutrinos without too more extra complexity, though. Can you suggest a set of parameters and consistency relations between them that would work for you, please?

**Comment by yymao** [comment]
@philbull --- I'm sorry for being slow here... but now I am confused again. If the validation function is provided by this package, why would other people (e.g. supernova people) need to write and test code to comply with the standard?

**Comment by philbull** [comment]
By "code to comply with the standard", I mean: if they write their own implementation of the CosmoBase class, it should comply with the standard [which is kind of a trivial comment]. They don't have to write their own code to _test_ whether it complies (that's what (C) provides), but they do have to do some work to make sure their implementation does comply.

**Comment by yymao** [comment]
Yes, of course. I guess the hope is that if they subclass the CosmoBase class, then it should be straightforward to be complying with the specification.

**Comment by philbull** [comment]
Yes, that should be very straightforward. I'm more worried about situations where people might create their own implementations from scratch though (e.g. because they're not using C/Python).

**Comment by c-d-leonard** [comment]
@tmcclintock I'm a bit confused - I thought that the way that cosmological observables are sensitive to the neutrino hierarchy was always via the sum of neutrino masses (in combination with the differences, from particle physics). Is there a way that cosmological observables are directly sensitive to individual masses? I could definitely be missing something here so please correct me.

I am also curious about the inclusion of both Omega_nu_mass / Omega_nu_rel and m_nu + N_nu_mass / N_nu_rel. Should there be / is there a consistency check to make sure these agree?

**Comment by philbull** [comment]
Yes, can the Omegas be derived from the other parameters in a straightforward way?

**Comment by tmcclintock** [comment]
@c-d-leonard I was actually hoping you could have hopped in and told me if I was wrong :). I don't know whether it's just the total that matters or not.
